### PR TITLE
Fixed #24923 -- error out nicely when using aggregates in order_by

### PR DESCRIPTION
**Pull request by akaariai**

### CODE REVIEW

File: tests/aggregation/tests.py [hunk 1]
```diff
@@ -105,6 +105,10 @@ def setUpTestData(cls):
     def test_empty_aggregate(self):
         self.assertEqual(Author.objects.all().aggregate(), {})
 
+    def test_aggregate_in_order_by(self):
+        with self.assertRaises(FieldError):
```
Comment (charettes) [inline tests/aggregation/tests.py]: I would test for the complete error message.

**Comment by charettes** [comment]
Commit message should be in past tense (`Errored out ...`) otherwise LGTM.

**Comment by jarshwah** [comment]
Do you think we should add another test that shows ordering by the reference to the aggregate annotation?

**Comment by akaariai** [comment]
No, I don't think we need a new test for aggregate annotation reference, this is tested already in multiple tests (for example aggregation tests, test_annotate_ordering.

File: django/db/models/sql/query.py [hunk 2]
```diff
@@ -1643,6 +1643,8 @@ def add_ordering(self, *ordering):
         for item in ordering:
             if not hasattr(item, 'resolve_expression') and not ORDER_PATTERN.match(item):
                 errors.append(item)
+            if getattr(item, 'contains_aggregate', False):
+                raise FieldError('Direct usage of aggregates in order_by is not allowed: %s' % item)
```
Comment (timgraham) [inline django/db/models/sql/query.py]: is this message an accurate improvement?
`Using an aggregate in order_by() without also including it in annotate() is not allowed:`

I'll also update the test to use `assertRaisesMessage` as suggested by Simon when merging.

**Comment by akaariai** [comment]
@timgraham yeah, your version of the message is accurate, and it is an improvement.

**Comment by timgraham** [comment]
Merged in 6f403056f09b0a77df9791125591f3057e328ab7.

**Pull request closed without merging**